beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go

```diff
@@ -54,3 +60,142 @@ func (vs *Server) getBellatrixBeaconBlock(ctx context.Context, req *ethpb.BlockR
 	blk.StateRoot = stateRoot
 	return blk, nil
 }
+
+// This function retrieves the payload header given the slot number and the validator index.
+// It's a no-op if the latest head block is not versioned bellatrix.
+func (vs *Server) getPayloadHeader(ctx context.Context, slot types.Slot, idx types.ValidatorIndex) (*ethpb.ExecutionPayloadHeader, error) {
+	if err := vs.BlockBuilder.Status(); err != nil {
+		return nil, err
+	}
+	b, err := vs.HeadFetcher.HeadBlock(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if blocks.IsPreBellatrixVersion(b.Version()) {
+		return nil, nil
+	}
+	h, err := b.Block().Body().ExecutionPayload()
+	if err != nil {
+		return nil, err
+	}
+	pk, err := vs.HeadFetcher.HeadValidatorIndexToPublicKey(ctx, idx)
+	if err != nil {
+		return nil, err
+	}
+	bid, err := vs.BlockBuilder.GetHeader(ctx, slot, bytesutil.ToBytes32(h.BlockHash), pk)
```

Comment:
if blockbuilder.Status() no longer requires the context because it's internal, do we need to pass the context here for get header?

Reply:
Yep! We'll need it for handling timeout which will come in the next PR : )